### PR TITLE
Transformations: Simple fix for labels to fields  problem with time series field names

### PR DESCRIPTION
**Pull request by torkelo**
Currently toDataFrame sets the field name to the series name instead of keeping the field name `Value`. 

Graphite & Prometheus already generates a series name based on the metric name and series name. this is then transformed using toDataFrame to a DataFrame. In this transform, we are using the series name as the field name for the value. So this is a big problem for the labels to fields transformer as we end up with this: 

Screenshot from 2020-04-22 09-04-45

So we have 2 ways to solve this problem either change so in toDateFrame when we set field name to 'Value': 
https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/dataframe/processDataFrame.ts#L76 

This has effects on other parts of Grafana where we generate series names on graphs and legends etc. 

The other solution, which this PR tries to explore is to fix it in the labels to fields transform. In this transform, I detect if a field is named 
the same as the data frame (series) and in that change the field to 'Value' so that many series can share the same Value column.

TODO

- [x] Fix unit test

### CODE REVIEW

File: packages/grafana-data/src/transformations/transformers/labelsToFields.ts [hunk 1]
```diff
@@ -64,6 +65,46 @@ export const labelsToFieldsTransformer: DataTransformerInfo<LabelsToFieldsOption
...
 
+function getFramesWithOnlyValueFields(data: DataFrame[]): DataFrame[] {
+  const processed: DataFrame[] = [];
+
+  for (const series of data) {
+    const fields: Field[] = [];
+
+    for (let i = 0; i < series.fields.length; i++) {
+      const field = series.fields[i];
+      if (field.type !== FieldType.number) {
```
Comment (ryantxu) [inline packages/grafana-data/src/transformations/transformers/labelsToFields.ts:76]: what about the time field?  Is that not wanted here?

Comment (torkelo) [reply packages/grafana-data/src/transformations/transformers/labelsToFields.ts:76]: not in this function, for some reason, the data is split into data frames with time fields and the data frames without, not sure why.

**Comment by torkelo** [comment]
Ok added unit test.

File: packages/grafana-data/src/transformations/transformers/labelsToFields.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-data/src/transformations/transformers/labelsToFields.ts:76]: yes, not splitting the frames like I did is one optimization we could do if we encounter perf issues with current solution.

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Pull request merged**